File: rpcgen/stdlib/ts/stdlib.ts

export class ByteReader {
    readInt32() { return 0 }
    readString() { return "" }
}
export class ByteWriter {
    writeInt32(x:number) {}
    writeString(x:string) {}

    asBuffer(): Buffer {
        return Buffer.alloc(42)
    }
}

export class Array<T> {
    constructor(
        private readonly mem: T[] = []
    ) {}

    size() {
        return this.mem.length
    }
    get(i:number) {
        return this.mem[i]
    }
    push(val:T) {
        this.mem.push(val)
    }
}

File: hehgen/stdlib/java/src/test/stdlib.java
package src.test;

import java.util.ArrayList;

class Array<T> {
    private final ArrayList<T> mem = new ArrayList<>();

    public T get(int i) {
        return mem.get(i);
    }
    public void push(T x) {
        mem.add(x);
    }
    public int size() {
        return mem.size();
    }
}

class ByteReader {
    int readInt32() { return 0; }
    String readString() { return ""; }
}
class ByteWriter {
    void writeInt32(int x) {}
    void writeString(String x) {}
}
